src/wire/packet_manager.cpp

```diff
@@ -769,6 +784,19 @@ bool PacketManager::ExecDescribeMessage(InputPacket *pkt) {
   return true;
 }
 
+// Execute the exec message in a worker thread
+void PacketManager::ExecExecuteMessageAsync(void* conn_ptr){
+  LOG_DEBUG("Exec Execute Message Async starting");
+  wire::LibeventSocket * conn = (wire::LibeventSocket *)conn_ptr;
+  InputPacket *pkt = &conn->rpkt;
+  PacketManager *pkt_mgr = &conn->pkt_manager;
+  pkt_mgr->ExecExecuteMessage(pkt, (size_t)conn->thread_id);
+  conn->worker_executing = false;
+  // tell the networking thread it can write
+  event_active(conn->event, EV_WRITE, 0);
+  LOG_DEBUG("Exec Execute Message Async done, activating the event");
+}
+
 void PacketManager::ExecExecuteMessage(InputPacket *pkt, const size_t thread_id) {
```

Comment:
I think we could remove ExecExecuteMessage() and ExecQueryMessage(), since these are no longer called.